File: Data Structures & Algorithms/append-characters-to-string-to-make-subsequence/submission-2.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        l,r=0,0
        while(l<len(t) and r<len(s)):
            if(t[l]==s[r]):
                l=l+1
            r=r+1
        return len(t)-l
        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
    TreeSet<Integer> set = new TreeSet<>();

    for (int num : nums) {
        set.add(num);
    }

    int max = 0;
    int streak = 0;
    int prev = Integer.MIN_VALUE;

    for (int current : set) {
        if (current == prev + 1) {
            streak++;
        } else {
            streak = 1;
        }

        max = Math.max(max, streak);
        prev = current;
    }

    return max;
}

}
